examples/reminders-react/src/api/useInfiniteCollection.ts

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import {
+	collection,
+	CollectionReference,
+	limit as limitConstraint,
+	onSnapshot,
+	orderBy,
+	query,
+	QueryConstraint,
+	Unsubscribe
+} from 'firebase/firestore';
+import { db } from './firebase';
+
+const useInfiniteCollection = <T extends { id: string }>(
```

Comment:
I see that you're using this `{ id: string }` on a couple of places. Consider creating a `BaseModel` type instead